sdk/core/dma/dma_compartment.cc

```diff
@@ -0,0 +1,196 @@
+#include "futex.h"
+#define MALLOC_QUOTA 0x100000
+
+#include "dma_compartment.hh"
+#include <cstdint>
+#include <memory>
+#include <debug.hh>
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <utils.hh>
+#include "platform-dma.hh"
+#include <errno.h>
+#include <interrupt.h>
+#include <locks.hh>
+
+// Expose debugging features unconditionally for this compartment.
+using Debug = ConditionalDebug<true, "DMA Compartment">;
+
+// Import some useful things from the CHERI namespace.
+using namespace CHERI;
+
+Ibex::PlatformDMA platformDma;
+
+DECLARE_AND_DEFINE_INTERRUPT_CAPABILITY(dmaInterruptCapability, dma, true, true);
+
+namespace {
+    /**
+     * Flag lock to control ownership 
+	 * over the dma controller
+     */
+	FlagLock dmaOwnershipLock;
+
+    uint32_t dmaIsLaunched = 0;
+    uint32_t previousInterruptCounter = 0;
+
+    /**
+     * Claims pointers so that the following 
+     * addresses will not be freed ahead of time
+     */
+    std::unique_ptr<char> claimedSource;
+    std::unique_ptr<char> claimedDestination;
+
+} // namespace
+
+int launch_dma(uint32_t *sourceAddress, uint32_t *targetAddress, uint32_t lengthInBytes,
+                        uint32_t sourceStrides, uint32_t targetStrides, uint32_t byteSwapAmount)
+{    
+
+	/** 
+     *  Before launching a dma, check whether dma
+	 *  is occupied by another thread, 
+     *  else lock the ownership and start the dma
+     *  via LockGuard.
+     * 
+     *  This lock automatically unlocks at the end of this function.
+     */	
+	LockGuard g{dmaOwnershipLock};
+
+    /**
+     *  If dma is already launched, we need to check for the interrupt status.
+	 *  No need for validity and permissions checks though for the scheduler 
+	 *  once futex is created   
+     */
+    
+	const uint32_t *dmaFutex = interrupt_futex_get(STATIC_SEALED_VALUE(dmaInterruptCapability));
+    uint32_t currentInterruptCounter = *dmaFutex;
+    
+    /** 
+     * sleep a bit, so that dmaIsLaunched can be flipped successfully
```

Comment:
What does this comment mean?

Reply:
I forgot to delete this, will delete it 👌